notify/notify.go

```diff
@@ -575,9 +575,13 @@ func (r RetryStage) Exec(ctx context.Context, alerts ...*types.Alert) (context.C
 
 		select {
 		case <-tick.C:
-			if err := r.integration.Notify(ctx, alerts...); err != nil {
+			if retry, err := r.integration.Notify(ctx, alerts...); err != nil {
 				numFailedNotifications.WithLabelValues(r.integration.name).Inc()
 				log.Warnf("Notify attempt %d failed: %s", i, err)
+				if !retry {
+					log.Warnf("Cancelling notify retry due to unrecoverable error: %s", err)
+					return ctx, alerts, nil
+				}
```

Comment:
We should probably make the first warning log a debug level log and show a final error after failing permanently. Either by no-retry or by timing out in retry (log last error then).